assets/blocks/reader-registration/index.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Newspack Blocks.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Blocks\ReaderRegistration;
+
+use Newspack;
+use Newspack\Reader_Activation;
+
+defined( 'ABSPATH' ) || exit;
+
+const FORM_ACTION = 'newspack_reader_registration';
+
+/**
+ * Do not register block hooks if Reader Activation is not enabled.
+ */
+if ( ! Reader_Activation::is_enabled() ) {
+	return;
+}
+
+/**
+ * Register block from metadata.
+ */
+function register_block() {
+	register_block_type_from_metadata(
+		__DIR__ . '/block.json',
+		array(
+			'render_callback' => __NAMESPACE__ . '\\render_block',
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\\register_block' );
+
+/**
+ * Enqueue front-end scripts.
+ */
+function enqueue_scripts() {
+	$handle = 'newspack-reader-registration-block';
+	\wp_enqueue_style(
+		$handle,
+		\Newspack\Newspack::plugin_url() . '/dist/reader-registration-block.css',
+		[],
+		NEWSPACK_PLUGIN_VERSION
+	);
+	\wp_enqueue_script(
+		$handle,
+		\Newspack\Newspack::plugin_url() . '/dist/reader-registration-block.js',
+		[ 'wp-polyfill', 'newspack-reader-activation' ],
+		NEWSPACK_PLUGIN_VERSION,
+		true
+	);
+	\wp_script_add_data( $handle, 'async', true );
+	\wp_script_add_data( $handle, 'amp-plus', true );
+}
+add_action( 'wp_enqueue_scripts', __NAMESPACE__ . '\\enqueue_scripts' );
+
+/**
+ * Render Registration Block.
+ *
+ * @param array[] $attrs Block attributes.
+ */
+function render_block( $attrs ) {
+	if ( \is_user_logged_in() || Reader_Activation::get_auth_intention_value() ) {
+		return;
```

Comment:
Empty render might be misleading, if the block is a part of a larger piece of UI (e.g. a prompt). I'd propose displaying the post-submission state by default ("Thank you for registering"), with a block attribute option to render empty if reader has identified themself.

Reply:
Among other improvements, 1e72eede288a658f0c0cd54dabf18cb3f17acc23 implements the "Thank you for registering!" message to authenticated, pre-authenticated and non-JS form requests.